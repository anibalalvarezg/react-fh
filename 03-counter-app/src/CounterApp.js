import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value } ) => {
    const [counter, setCounter] = useState(value ? value : 0);

    const handleAdd = () => setCounter(counter + 1);
    const handleSubtract = () => setCounter(counter - 1);
    const handleReset = () => setCounter(0);

    return (
        <>
            <h1> Counter App </h1>
            <p> { counter } </p>

            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleSubtract }>-1</button>
            <button onClick={ handleReset }>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number,
};

CounterApp.defaultProps = {
    value: 0,
};

export default CounterApp;