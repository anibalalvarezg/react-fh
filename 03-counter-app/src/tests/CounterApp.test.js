import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Tests CounterApp component', () => {
    test('should show CounterApp correctly', () => {
        const wrapper = shallow(<CounterApp />); 
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should show value by default 100', () => {
        const wrapper = shallow( <CounterApp value={100}/>);
        const counterText = wrapper.find('p').text().trim();

        expect(counterText).toBe('100'); 
    });
    
});