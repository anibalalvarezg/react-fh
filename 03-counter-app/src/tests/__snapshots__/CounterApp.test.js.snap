// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests CounterApp component should show CounterApp correctly 1`] = `
<Fragment>
  <h1>
     Counter App 
  </h1>
  <p>
     
    0
     
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
